pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -672,6 +673,42 @@ func (m *kubeGenericRuntimeManager) killContainersWithSyncResult(pod *v1.Pod, ru
 	return
 }
 
+func (m *kubeGenericRuntimeManager) preStopSidecars(pod *v1.Pod, containers []*kubecontainer.Container) {
```

Comment:
the function signature seems confusing as this can be used to preStop whatever slice of containers. I'd suggest to make it generic, aka without the assumption that the container slice are all sidecars, or subsequent contributors can get confused

Reply:
agreed changed